projects/controllable_dialogue/README.md

```diff
@@ -24,11 +26,254 @@ improvements in human quality judgments.
+    --fromfile-datapath data/ConvAI2_controllable/train.txt \
+    --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+    --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+    -mf /path/to/your/modelfile \
+    --control-vars avg_nidf
+
+Note: if you add your paths for `train.txt`, `valid.txt` and `dict_twit30k_train_split` to `train_controllable_seq2seq.py` then you won't need to enter them via command line.
+
+This will default to embedding size 10, but you could include e.g. `--control-embeddingsize 15` if you wanted to change it.
+
+This code will also default to 10 NIDF buckets. If you want to use a different number of buckets, first you need to figure out what the NIDF lower bound should be for each bucket. Suppose you want 8 buckets. First run:
+
+    python projects/controllable_dialogue/get_bucket_lowerbounds.py \
+      --fromfile_datapath data/ConvAI2_controllable/train.txt \
+      --num_buckets 8 \
+      --control avg_nidf
+
+and then copy and paste the provided lower bounds into `controllable_dialogue/controllable_seq2seq/controls.py`, similarly to the existing `AVG_NIDF_10BUCKET_LBS`. Then you can train a model with `--control-num-buckets 8`.
+
+You can train a CT model conditioned on multiple controls:
+
+    python projects/controllable_dialogue/train_controllable_seq2seq.py \
+      --fromfile-datapath data/ConvAI2_controllable/train.txt \
+      --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+      --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+      -mf /path/to/your/modelfile \
+      --control-vars avg_nidf,question
+
+To take an existing non-CT model (e.g. the baseline) and the finetune it as a CT model do this:
+
+    python projects/controllable_dialogue/train_controllable_seq2seq.py \
+    --fromfile-datapath data/ConvAI2_controllable/train.txt \
+    --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+    --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+    -mf /path/to/your/modelfile \
+    --init-model data/models/controllable_dialogue/convai2_finetuned_baseline \
+    --add-control True \
+    -cv avg_nidf
+
+This command will take the parameters saved in `--init-model`, load them in the new model (which has randomly initialized weights for the new CT parameters), and then save that model to the given modelfile (`-mf`). It should be quick. Once that's done, run this command:
+
+    python projects/controllable_dialogue/train_controllable_seq2seq.py \
+    --fromfile-datapath data/ConvAI2_controllable/train.txt \
+    --fromfile-datapath2 data/ConvAI2_controllable/valid.txt \
+    --dict-file data/models/controllable_dialogue/dict_twit30k_train_split \
+    -mf /path/to/your/modelfile \
+    --add-control False \
+    -cv avg_nidf
+
+You should see your new CT model training. Note: this is how the models in the paper were trained.
+
+## Look at output and automatic metrics
```

Comment:
I'd like to add an integration test which computes automatic metrics for some of these models and checks nightly that they produce the right results. This helps us support models for a long time.

For that we would need to collect what are the true metrics

An example which already runs the test nightly:
https://github.com/facebookresearch/ParlAI/blob/52364b2466124561138a23fc53fd3061118ac1b4/tests/nightly/gpu/test_wizard.py#L48-L61